src/modules/ZFCPController.rb

```diff
@@ -720,6 +702,100 @@ def GetLUNs(busid, wwpn)
     publish function: :DeactivateDisk, type: "void (string, string, string)"
     publish function: :GetWWPNs, type: "list <string> (string)"
     publish function: :GetLUNs, type: "list <string> (string, string)"
+
+  private
+
+    # Finds the activated controllers
+    #
+    # Initially, it reads the activated controllers from hwinfo.
+    #
+    # @return [Array<String>] List of controller channels
+    def activated_controllers
+      return @activated_controllers if @activated_controllers
+      ctrls = GetControllers().select do |ctrl|
+        io = ctrl.fetch("resource", {}).fetch("io", []).first
+        io && io["active"]
+      end
+      log.info "Already activated controllers: #{ctrls}"
+      @activated_controllers = ctrls.map { |c| c["sysfs_bus_id"] }
+    end
+
+    # Mark a controller as activated
+    #
+    # @param channel [String] Channel
+    def register_as_activated(channel)
+      @activated_controllers << channel
```

Comment:
just for safety and more robust api I would call here instead `activated_controllers << channel` which ensures that activated_controllers is not nil.